cargo-dist/src/backend/installer/homebrew.rs

```diff
@@ -178,9 +182,64 @@ pub fn to_class_case(app_name: &str) -> String {
     chars.iter().collect()
 }
 
+/// Converts SPDX license string into Homebrew Ruby DSL
+// Homebrew DSL reference: https://docs.brew.sh/License-Guidelines
+pub fn to_homebrew_license_format(app_license: &str) -> Result<String, ParseError> {
+    let spdx = Expression::parse(app_license)?;
+    let mut spdx = spdx.iter().peekable();
+    let mut buffer: Vec<String> = vec![];
+
+    while let Some(token) = spdx.next() {
+        match token {
+            ExprNode::Req(req) => {
+                // If token is a license, push to the buffer as-is for next operator or end.
+                let requirement = format!("\"{}\"", req.req);
+                buffer.push(requirement);
+            }
+            ExprNode::Op(op) => {
+                // If token is an operation, group operands in buffer into all_of/any_of clause.
+                // Operations are postfix, so we pop off the previous two elements and combine.
+                let second_operand = buffer.pop().expect("Operator missing first operand.");
```

Comment:
Are these cases unreachable, or could they happen with a malformed SPDX string?

Reply:
They _should_ be unreachable, as `Expression::parse()` will return an error and early-out if the SPDX string is malformed. I added error messages here anyways because the invariant (operators are always preceded by two operands) is not statically enforced. I considered returning a ParseError, but it gets a bit icky because instantiating `spdx`'s ParseError requires a "span" of which characters in the string were unable to be parsed, and only the license tokens keep track of their span, not the operators.

Reply:
Basically SPDX takes a string like `"A OR B"` and then creates a stream like `[Req(A), Req(B), Op(Or)]`, so there is no valid SPDX string that would result in an operator without two operands. If this panic occurs, it means there is a serious logic issue in either this code or in the `spdx` crate. I have tested a couple weird strings that abstractly correspond to this case, e.g. "Apache-2.0 OR" or "MIT AND OR", but both properly return ParseErrors when parsed by spdx (I'll add these "invalid" test cases along with homebrew's cases mentioned below).

Reply:
Perfect, thank you! I appreciate you testing this out to confirm it.